From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Sat, 26 Dec 2020 19:21:35 +0100
Subject: [PATCH] {PRINTF} kernel/printk/printk: add __printf attribute to
 printk_sprint
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

gcc on x86-64 reports:

    kernel/printk/printk.c: In function ‘printk_sprint’:
    kernel/printk/printk.c:1912:2: error: function ‘printk_sprint’ might
    be a candidate for ‘gnu_printf’ format attribute
    [-Werror=suggest-attribute=format]
     1912 |  text_len = vscnprintf(text, size, fmt, args);
          |  ^~~~~~~~
---
 kernel/printk/printk.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/printk/printk.c b/kernel/printk/printk.c
index 825277e1e742..363a072c66ed 100644
--- a/kernel/printk/printk.c
+++ b/kernel/printk/printk.c
@@ -2066,7 +2066,7 @@ u16 printk_parse_prefix(const char *text, int *level,
 	return prefix_len;
 }
 
-static u16 printk_sprint(char *text, u16 size, int facility,
+static __printf(5, 0) u16 printk_sprint(char *text, u16 size, int facility,
 			 enum printk_info_flags *flags, const char *fmt,
 			 va_list args)
 {
-- 
